### DISCUSSION TITLE
useFieldArray default keyName `id` => `key`?

### DISCUSSION THREAD

**Discussion by viktor89** [react-hook-form/react-hook-form#6962] [Ideas] [answered]
Often times I run into the issue that useFieldArray defaults to keyName='id'. This is an issue because most times the objects in the field arrays have their own id-field - especially if you're editing some existing data, rather than creating new.

This is of course solved by changing the keyName in `useFieldArray`. I do however think the default naming is bad. Since this value is used as a `key`-prop, shouldn't it rather be called `key` by default? This would make more sense, IMO. It's not a big thing, but I think you would see less need for changing the default `keyName` if it wasn't defaulted to `'id'`.

**Comment by bluebill1049** [comment, marked as answer]
This is something we could consider in the next breaking change (major version), in fact, react 18 will have a new hook `useId`, we can look into that later on as well.
